packages/gp2-components/src/organisms/__tests__/ResourceModal.test.tsx

```diff
@@ -27,21 +41,42 @@ const enterDescription = (description: string) =>
   userEvent.type(descriptionBox(), description);
 
 const linkBox = () => screen.getByRole('textbox', { name: /url/i });
-const enterLink = (link: string) => userEvent.type(linkBox(), link);
+const enterLink = (link: string) => {
+  userEvent.clear(linkBox());
+  userEvent.type(linkBox(), link);
+};
+
+let getUserConfirmation!: jest.MockedFunction<
+  (message: string, callback: (confirmed: boolean) => void) => void
+>;
+let history!: History;
+
 beforeEach(jest.resetAllMocks);
 
 describe('ResourceModal', () => {
-  beforeEach(() => {
-    render(<ResourceModal {...defaultProps} />, { wrapper: StaticRouter });
-  });
+  const renderResourseModal = (
+    props: Partial<ComponentProps<typeof ResourceModal>> = {},
+  ) => {
+    render(<ResourceModal {...defaultProps} {...props} />, {
+      wrapper: StaticRouter,
+    });
+  };
   test('title and description should be disabled and url is hidden', () => {
+    renderResourseModal();
     expect(descriptionBox()).toBeDisabled();
     expect(titleBox()).toBeDisabled();
     expect(
       screen.queryByRole('textbox', { name: /url/i }),
     ).not.toBeInTheDocument();
   });
+  it('see if the modal is for adding resource', () => {
```

Comment:
we should pass the props we expect them to have a check for those, I don't like the title either

Reply:
Done